### PR TITLE
[feature] introduce cyrus_sasl_saslauthd_flags

### LINKED ISSUE #9: the role does not manage options of saslauthd
the role does not support extra flags for the service command.

### PR DESCRIPTION
**Pull request by trombik**
fixes #9

### CODE REVIEW

**Comment by trombik** [comment]
i think that the variable name is horrible. as this is a common implementation among almost all roles, better, standard name for the variable, and the rule in `ansible-role-example`, should be created.

**Review by mitsururike — COMMENTED** [review]
i do not understand the necessity of `cyrus_sasl_saslauthd_extra_variables_default`.
if it is not supposed to be modified by users, it should be notified in README.md.

**Comment by trombik** [comment]
IMO, simply call it `foo_flags` and `foo_flags_default` is better like https://github.com/reallyenglish/ansible-role-rabbitmq/blob/master/README.md#rabbitmq_flags

**Comment by trombik** [comment]
> i do not understand the necessity of cyrus_sasl_saslauthd_extra_variables_default.

without it, users have to set `cyrus_sasl_saslauthd_extra_variables` explicitly in their role variable and the defaults set by the upstream would be removed, which would surprise users. why is it a role variable? because, even when we are unable to catch up with changes in the default values, which is very likely, users are able to workaround it without modifying tasks or anything in the role.

**Comment by trombik** [comment]
in our use case, without it, you would have to modify files under `roles.galaxy`, which is a big no-no unless you are trying to create a patch and test it in the real use case. you would have two options: violating the rule, or wait for the issue to get fixed.

**Comment by trombik** [comment]
`foo_flags` does not describe itself the best, either, but i cannot think of any other common denominator in `/etc/default/foo`, `/etc/sysconfig/foo` and `/etc/rc.conf.d/foo`.

**Comment by trombik** [comment]
created https://github.com/reallyenglish/ansible-role-example/pull/120

**Comment by mitsururike** [comment]
> without it, users have to set cyrus_sasl_saslauthd_extra_variables explicitly in their role variable and the defaults set by the upstream would be removed, which would surprise users. why is it a role variable? because, even when we are unable to catch up with changes in the default values, which is very likely, users are able to workaround it without modifying tasks or anything in the role.

it is a good explanation to know the necessity of the variable and how to use it.
it should be written in the README.md.
at least,
* *_flags_default keeps the defaults set by the upstream
* when users should use *_flags_default variable
* *_flags and *_flags_default is merged

should be described.

**Comment by trombik** [comment]
your argument includes two different suggestions, one for users and others for developers. of course, user should primarily use `cyrus_sasl_saslauthd_flags`. but if a user want to shoot in the foot, so be it. the latter one has been discussed and solved, so I believe, in other PR.

**Comment by mitsururike** [comment]
which do you mean "`*_flags_default` must not be modified by users" or "`*_flags_default` should be modified by users for workaround against the upstream changes"?
either way, it should be described in README.md.

File: README.md [hunk 1]
```diff
@@ -50,7 +50,9 @@ None
 
 This variable is a dict of variables of configuration files, such as
 `/etc/default/saslauthd`. A key in the dict is name of the variable in the
-file, and the value of the key is value of the variable.
+file, and the value of the key is value of the variable. The variable is
+combined with `cyrus_sasl_saslauthd_flags_default`, which includes upstream's
```
Comment (mitsururike) [inline README.md]: `include` means "have something as a part of if". to clarify nothing other than the default values are included, "keeps" (or something similar word) is appropriate.

**Review by mitsururike — APPROVED** [review]

**Pull request merged**